server/src/test/java/org/apache/druid/segment/realtime/appenderator/StreamAppenderatorTest.java

```diff
@@ -2546,6 +2546,65 @@ public void test_dropSegment_skipsUnlockInterval_ifOverlappingSinkIsActive() thr
     }
   }
 
+  /**
+   * Tests an edge case where a writable sink exists but has no hydrants.
+   * <p>
+   * The test scenario is somewhat unrealistic, but we are trying to prevent a regression because this issue was seen in
+   * the wild and resulted in failed indexing tasks due to an IAE.
+   * </p>
+   */
```

Comment:
I think the javadoc here can be omitted.
Also, since it was already reported, I think it is a viable scenario.